gapis/api/vulkan/overdraw.go

```diff
@@ -0,0 +1,1331 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vulkan
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/core/memory/arena"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/transform"
+	"github.com/google/gapid/gapis/capture"
+	"github.com/google/gapid/gapis/memory"
+	"github.com/google/gapid/gapis/messages"
+	"github.com/google/gapid/gapis/replay"
+	"github.com/google/gapid/gapis/service"
+	"github.com/google/gapid/gapis/service/path"
+)
+
+var (
+	_ = transform.Transformer(&stencilOverdraw{})
+)
+
+// TODO: add support for specific renderpass in a queue submit
+type stencilOverdraw struct {
+	rewrite map[api.CmdID]replay.Result
+}
+
+func newStencilOverdraw() *stencilOverdraw {
+	return &stencilOverdraw{
+		rewrite: map[api.CmdID]replay.Result{},
+	}
+}
+
+func (s *stencilOverdraw) add(ctx context.Context, extraCommands uint64, after uint64, capt *path.Capture, res replay.Result) {
+	c, err := capture.ResolveFromPath(ctx, capt)
+	if err != nil {
+		res(nil, err)
+	}
+	// TODO: Ideally this would be smarter, but without duplicating the
+	// state and mutating it, it's hard to tell what the right
+	// vkQueueSubmit to modify is.
+	lastSubmit := int64(after)
```

Comment:
This logic should be improved to allow selecting a specific renderpass without counting on the early terminator.  #2010 needs to be resolved first though.